 0016-capi-upgrades/0016-capi-upgrades.md

```diff
@@ -0,0 +1,49 @@
+# RFC 016 - CLUSTER API UPGRADES
+
+## Reasoning
+
+As part of the Cluster API (CAPI) Hive Sprint, the `upgrades hive` decided to create a document to gather all actions need to be done in order to upgrade a workload cluster successfully as well as bring the functionality we have already built to trigger the upgrades.
+
+## Scope
+
+This document covers only the action to upgrade the workload cluster without solving problems related to the versioning of the different CAPI controllers and/or deciding which controller reconciles a cluster.
+
+At the same time all the automation on top of the upgrades, like freeze periods or maintainance windows, are out of the scope here ([include in RFC 0015](/0015-automatic-cluster--upgrades)).
+
+Furthermore, a workload cluster can be upgraded in different ways:
+
+- Apps installed in the cluster are upgraded (out of scope here). We have other non-CAPI components that already take care of that, like app-operator
+
+- Cluster is reconciled by a newer CAPI controller. We may want to upgrade the Cluster to a newer CAPI controller release, which may change the infrastructure created to support the cluster (network, load balancers, etc). We do this update just changing the CAPI label on the Cluster CR, and we don't need a new machine image to be created/referenced by the CR's.
```

Comment:
A new CAPI controller does not mean any change on anything. In theory, a controller change should never touch existing resources. We still want to do this in a controlled manner, as we want to reduce the risk of accidental changes as well as create full reproducibility between an existing cluster and a new one.

Reply:
but do we know if you introduce a new controller that differs how infra is configured, the controller will reconcile the existing resources to be like the new controller logic? cc/ @fiunchinho

Reply:
IMHO we should do it like upstream and try to not change infra just by having a new controller version, but do it more specified